wgpu-native/src/conv.rs

```diff
@@ -287,10 +287,69 @@ pub fn map_texture_format(texture_format: resource::TextureFormat) -> hal::forma
     use hal::format::Format as H;
     use crate::resource::TextureFormat::*;
     match texture_format {
-        R8g8b8a8Unorm => H::Rgba8Unorm,
-        R8g8b8a8Uint => H::Rgba8Uint,
-        B8g8r8a8Unorm => H::Bgra8Unorm,
+        // Normal 8 bit formats
+        R8Unorm => H::R8Unorm,
+        R8UnormSrgb => H::R8Srgb,
+        R8Snorm => H::R8Inorm,
+        R8Uint => H::R8Uint,
+        R8Sint => H::R8Int,
+
+        // Normal 16 bit formats
+        R16Unorm => H::R16Unorm,
+        R16Snorm => H::R16Inorm,
+        R16Uint => H::R16Uint,
+        R16Sint => H::R16Int,
+        R16Float => H::R16Float,
+
+        Rg8Unorm => H::Rg8Unorm,
+        Rg8UnormSrgb => H::Rg8Srgb,
+        Rg8Snorm => H::Rg8Inorm,
+        Rg8Uint => H::Rg8Uint,
+        Rg8Sint => H::Rg8Int,
+
+        // Packed 16 bit formats
+        B5g6r5Unorm => H::B5g6r5Unorm,
```

Comment:
Should this be `H::R5g6b5Unorm`? I forget whether packed 16-bit format names are reversed like packed 32-bit formats

Reply:
Hah, right. Some of our formats are "packed" (there is a `bool` describing that in the format description), which means on little endian systems the order of components is reverse.

Reply:
for example:
```rust
R5_G6_B5 { !16, COLOR, (1, 1), color: 16 },
```
the `!` before the bit count means that it's a packed format

We need to clarify with WebGPU folks if the packing semantics is the same there

Reply:
Yeah, they're reversed in the `Packed 32 bit formats` mapping here (which I think is correct), I think this one just needs to be reversed too.

Agreed we should clarify in WebGPU though. Because we're using Apple's names then I assume that's the intent (i.e. all packed formats are reversed).

Reply:
Just filed https://github.com/gpuweb/gpuweb/issues/224
let's proceed here until upstream is clarified, and then fix if needed